lib/apiClient.ts

```diff
@@ -0,0 +1,64 @@
+import fetch, { HeadersInit, RequestInit } from "node-fetch";
+import { ApiError } from "./errors";
+import { bodyToCamelCase, getUserAgent } from "./utils";
+
+export class ApiClient {
+  private _apiKey: string;
+  private _apiVersion: string;
+
+  constructor(apiKey: string, apiVersion: string = "v2beta") {
+    this._apiKey = apiKey;
+    this._apiVersion = apiVersion;
+  }
+
+  private _url(path: string): string {
+    return `https://api.bit.io/${this._apiVersion}${path}`;
+  }
+
+  private get _headers(): HeadersInit {
+    return {
+      "Accept": "application/json",
+      "Authorization": `Bearer ${this._apiKey}`,
+      "User-Agent": getUserAgent(),
+    };
+  }
+
+  async request<T>(
+    method: string,
+    path: string,
+    init?: RequestInit,
+  ): Promise<T> {
+    const response = await fetch(this._url(path), {
+      ...init,
+      headers: {
+        ...init?.headers,
+        ...this._headers,
+      },
```

Comment:
maybe doesn't matter, but should User-Agent header be overrideable?

Reply:
No I think it's useful for our analytics purposes to force the UA here